iree/turbine/kernel/wave/compile_options.py

```diff
@@ -58,6 +58,7 @@ class WaveCompileOptions:
     )
     use_local_scope: bool = False
     use_water_leak_check: bool = False
+    optimization_level: int = 2
```

Comment:
Nit 1: this look like it doesn't belong under the "debug options" comment below.
Nit 2: I'd expect the optimization level for a compiler to be 0. There is also no differentiation in the code between 1 and 2, all conditions are `> 0`, so unclear why 2 specifically is chosen.

Reply:
I agree, since we don't have a sense for what the different levels should be, maybe keeping it boolean makes sense. I think the canonicalize change might be a big one, so I am okay with splitting it up into 2 PRs, if that is the case.